nomad/state/events.go

```diff
@@ -179,15 +175,11 @@ func eventFromChange(change memdb.Change) (structs.Event, bool) {
 			return structs.Event{}, false
 		}
 
-		// Node secret ID should not be included
-		node := after.Copy()
-		node.SecretID = ""
-
 		return structs.Event{
 			Topic: structs.TopicNode,
-			Key:   node.ID,
+			Key:   after.ID,
 			Payload: &structs.NodeStreamEvent{
-				Node: node,
+				Node: after,
```

Comment:
@schmichael , @tgross , do you think it's worth starting and documenting a `func ... Sanitize()` pattern?

Reply:
Yeah, I think the general pattern is something like:

```go
func (t *T) Sanitzie() *T {
  t2 := t.Copy()
  t2.Secrets = ""
  return t2
}
```

Now if we're really clever we'd figure out a way to use different types for these to let the compiler help us assert the right form is going the right place.... but I'm not sure that's tenable. I think the best it could look like is https://play.golang.org/p/33SEO_o4iG1

Which requires changing *a lot* of code, so I don't think that's going to be worth it.